storage/client.go

```diff
@@ -131,7 +143,33 @@ func NewBasicClient(accountName, accountKey string) (Client, error) {
 		return NewEmulatorClient()
 	}
 	return NewClient(accountName, accountKey, DefaultBaseURL, DefaultAPIVersion, defaultUseHTTPS)
+}
+
+// NewBasicClientChina constructs a Client with given storage service name and
+// key in the China cloud.
+func NewBasicClientChina(accountName, accountKey string) (Client, error) {
+	if accountName == StorageEmulatorAccountName {
+		return NewEmulatorClient()
+	}
+	return NewClient(accountName, accountKey, ChinaBaseURL, DefaultAPIVersion, defaultUseHTTPS)
+}
 
+// NewBasicClientGermany constructs a Client with given storage service name and
+// key in the Germany cloud.
+func NewBasicClientGermany(accountName, accountKey string) (Client, error) {
```

Comment:
I'd prefer to not duplicate these methods in order to support different clouds. What about using something like:

``` Go
func NewBasicClient(accountName, accountKey string) {
    return NewBasicClientc(accountName, accountKey, DefaultBaseURL)
}

func NewBasicClientc(accountName, accountKey, cloud string) {
    if accountName == StorageEmulatorAccountName {
        return NewEmulatorClient()
    }
    return NewClient(accountName, accountKey, cloud, DefaultAPIVersion, defaultUseHTTPS)
}
```

Also, I'm not married to the idea of using the name `NewClientc` I'd definitely love to come up with something better!